src/sentry/eventstream/base.py

```diff
@@ -188,3 +215,13 @@ def run_post_process_forwarder(
     ) -> None:
         assert not self.requires_post_process_forwarder()
         raise ForwarderNotRequired
+
+    @staticmethod
+    def _get_event_type(event: Event | GroupEvent) -> EventStreamEventType:
+        if getattr(event, "occurrence", None):
```

Comment:
ideally long term we'd just wrap all this in `get_event_type()`

Reply:
Do you mean in `event.get_event_type` or somewhere else?